src/allmydata/test/web/test_status.py

```diff
@@ -0,0 +1,152 @@
+# Tests for code in allmydata.web.status
+
+from bs4 import BeautifulSoup
+from twisted.web.template import flattenString
+from zope.interface import implementer
+
+from allmydata.interfaces import (
+    IDownloadResults,
+    IDownloadStatus,
+)
+from allmydata.web.status import DownloadStatusElement
+
+from .common import (
+    assert_soup_has_favicon,
+    assert_soup_has_text,
+    assert_soup_has_tag_with_content,
+)
+from ..common import TrialTestCase
+
+
+@implementer(IDownloadResults)
+class FakeDownloadResults(object):
+
+    def __init__(self,
+                 file_size=0,
+                 servers_used=None,
+                 server_problems=None,
+                 servermap=None,
+                 timings=None):
+        """
+        See IDownloadResults for parameters.
+        """
+        self.file_size = file_size
+        self.servers_used = servers_used
+        self.server_problems = server_problems
+        self.servermap = servermap
+        self.timings = timings
+
+
+@implementer(IDownloadStatus)
+class FakeDownloadStatus(object):
```

Comment:
The real download status looks fairly straightforward to instantiate.  Can it be used here instead of a fake?  I saw your note that its `get_results` returns `None` but maybe just that one aspect of it can be faked rather than the entire thing.

Reply:
Good idea, thank you!

Made this change in c14197e: `FakeDownloadStatus` is a `DownloadStatus` with a `get_results()` that returns some fake data.